flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerITCase.java

```diff
@@ -53,7 +53,12 @@
 import static org.junit.Assert.assertThat;
 import static org.junit.Assert.fail;
 
-/** IT cases for the {@link FlinkKafkaProducer}. */
+/**
+ * IT cases for the {@link FlinkKafkaProducer}.
+ *
+ * <p>Do not run this class in the same junit execution with other tests in your IDE. This may lead
+ * leaking threads.
```

Comment:
Is this an issue because tests may be run in parallel in the same JVM of the IDE and thus the leak detection looks at the threads of another test?

Reply:
If we see more tests than that, we could also introduce an annotation later.

Reply:
> Is this an issue because tests may be run in parallel in the same JVM of the IDE and thus the leak detection looks at the threads of another test?

Are tests within the same module by default run in parallel? Perhaps there is also another test within the module leaking thread that is executed before the `FlinkKafkaProducerITCase` ...

Reply:
> If we see more tests than that, we could also introduce an annotation later.

Good idea, hopefully, this case does not become more often